contracts/interfaces/WhitelistInterface.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.6.10;
+
+interface WhitelistInterface {
+    function whitelistProduct(
+        address underlying,
+        address strike,
+        address collateral
+    ) external returns (bytes32 id);
+
+    function whitelistCollateral(address collateral) external returns (bytes32 id);
+
+    function isWhitelistedProduct(
+        address underlying,
+        address strike,
+        address collateral
+    ) external view returns (bool);
+
+    function isWhitelistedCollateral(address collateral) external view returns (bool);
+
+    function registerOtoken(address oToken) external;
```

Comment:
In the implementatio & spec I changed it to `function whitelistOtoken(address _oToken) external` so we keep using same "whitelist" and "isWhitelisted" terminology.

Reply:
ahh that's good!

Reply:
@haythem96  @aparnakr 
I think for oTokens, it actually make more sense to call it `registerOtoken`. because we're whitelisting some tokens out of all the oTokens, we literally just record every token created by our own factory. So it's more like a record of "tokens from our factory" than "tokens that we whitelisted".

Reply:
IMO there is not that big difference, registering all otokens or whitelisting them is the same, better to use same wording everywhere I think.